client/client.go

```diff
@@ -19,6 +19,7 @@ type Client struct {
 	engine              engine.Engine // The core business logic of the client
```

Comment:
@lalexgap Can this commit (1a151dd263b63b9449949599acba63f4da258899) be turned into a standalone PR?

Reply:
Yup, I think it could be. Although I'm not super happy about having logic sprinkled everywhere to refetch the objective.